lib/pbench/server/api/resources/graphql_api.py

```diff
@@ -0,0 +1,50 @@
+import requests
+from flask_restful import Resource, abort
+from flask import request, make_response
+from flask_cors import cross_origin
+
+
+class GraphQL(Resource):
+    """GraphQL API for post request via server."""
+
+    def __init__(self, graphql_uri, logger, api):
+        self.graphql = graphql_uri
+        self.api = api
+        self.logger = logger
+
+    @cross_origin()
+    def post(self):
+        json_data = request.get_json(silent=True)
+
+        if not json_data:
+            self.logger.warning("GraphQL: Invalid json object {}", request.url)
+            abort(400, message="GraphQL: Invalid json object in request")
+
+        try:
+            # query GraphQL
+            gql_response = requests.post(self.graphQL, json=json_data)
+        except requests.exceptions.ConnectionError:
+            self.logger.exception("Connection refused during the GraphQL post request")
+            abort(502, message="Network problem, could not post to GraphQL Endpoint")
+        except requests.exceptions.Timeout:
+            self.logger.exception(
+                "Connection timed out during the GraphQL post request"
+            )
+            abort(
+                504, message="Connection timed out, could not post to GraphQL Endpoint"
+            )
+        except Exception:
+            self.logger.exception("Exception occurred during the GraphQL post request")
+            abort(500, message="INTERNAL ERROR")
+
+        try:
+            # construct response object
+            response = make_response(gql_response.text)
+        except Exception:
+            self.logger.exception("Exception occurred GraphQL response construction")
+            abort(
+                500, message="INTERNAL ERROR",
```

Comment:
I wonder why this formats differently from line 38?

Reply:
Ahh I thinks its because of the `,` at the end, good catch